### ISSUE TITLE
ON_ELEMENT_SELECTED routing broken in latest release

### ISSUE THREAD

**Issue by IGx89** [jovotech/jovo-framework#346]
## I'm submitting a...

- [x] Bug report
- [ ] Feature request
- [ ] Documentation issue or request
- [ ] Other... Please describe:

## Expected Behavior
An Alexa request of type "Display.ElementSelected" should trigger the "ON_ELEMENT_SELECTED" handler in a Jovo app.

## Current Behavior
ON_ELEMENT_SELECTED is not triggered

## Analysis
This was broken by #345, though the root problem appears to have been caused by https://github.com/jovotech/jovo-framework/commit/bcc94ac81642d93b3f4f3fa31e8646154e6e2e5d. It's caused by calls to middleware('before.XXX") and middleware("after.XXX") being always normalized to just middleware("XXX"), so the "before.router" code added in Handler.js as part of #345 ends up running after the main "router" code. That causes handleRequest.jovo.$handlers to always be undefined when Router.router runs.

## Workaround
Put the following in your app.js right after the call to `new App()`
```
app.middleware('router').fns = app.middleware('router').fns.reverse();
```

## Your Environment

* Jovo Framework version used: 2.0.8
* Operating System: Windows

**Comment by IGx89** [comment]
CC @aswetlow

**Comment by aswetlow** [comment]
Thanks, Matthew! Fixed in 2.0.9
